src/server/game/Spells/Auras/SpellAuras.cpp

```diff
@@ -2039,6 +2042,46 @@ void Aura::TriggerProcOnEvent(uint8 procEffectMask, AuraApplication* aurApp, Pro
         Remove();
 }
 
+void Aura::HeartbeatResistance(uint32 diff)
+{
+    m_heartBeatTimer += diff;
+
+    if (m_heartBeatTimer < 950)
+        return;
+
+    Unit* target = GetOwner()->ToUnit();
+    Unit* caster = GetCaster();
+
+    if (!target || !caster)
+        return;
+
+    if (target->GetTypeId() == TYPEID_UNIT && caster->GetTypeId() == TYPEID_PLAYER)
+    {
+        m_heartBeatTimer = 0;
+
+        uint32 auraTimePassed = time(nullptr) - GetApplyTime();
```

Comment:
why do you tick every second if you only want to check three times per aura? why not just set `m_heartBeatTimer` to whenever the first check should be in aura apply, and then reset to that value after each heartbeat?